tests/unit/SubmissionsApiTest.php

```diff
@@ -0,0 +1,434 @@
+                )
+            )
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->responseMock->expects(self::any())->method('json')->willReturn(json_decode($body, true));
+        $this->responseMock->expects(self::any())->method('getBody')->willReturn($body);
+
+        $this->responseMock->expects($this->any())->method('getStatusCode')->willReturn($code);
+    }
+
+    /**
+     * @covers \Smartling\Submissions\SubmissionsApi::createSubmission
+     */
+    public function testCreateSubmission()
+    {
+
+        $testRequestBody = [
+            'original_asset_id' => ['a' => '1'],
+            'title' => 'Submission 1',
+            'fileUri' => '/posts/hello-world_1_1_post.xml',
+            'original_locale' => 'en-US'
+        ];
+
+        $testRawResponse = json_encode(
+            [
+                "response" => [
+                    "code" => "SUCCESS",
+                    "data" => [
+                        "submission_uid" => "8264fd9133d3",
+                        "project_id" => $this->projectId,
+                        "bucket_name" => "name",
+                        "original_asset_id" => ["a" => "1"],
+                        "title" => "Submission 1",
+                        "fileUri" => "/posts/hello-world_1_1_post.xml",
+                        "total_word_count" => "0",
+                        "total_string_count" => "0",
+                        "content_hash" => null,
+                        "original_locale" => "en-US",
+                        "outdated" => "0",
+                        "last_modified" => null,
+                        "custom_original_data" => null,
+                        "createdAt" => "2018-07-31T10:37:20.839Z",
+                        "updatedAt" => "2018-07-31T10:37:20.839Z",
+                        "details" => []
+                    ]
+                ]
+            ]
+        );
+
+        $this->mockClientResponse(200, $testRawResponse);
+
+        $testExpectedResponse = json_decode($testRawResponse, true)['response']['data'];
+
+        $bucketName = 'name';
+        $endpointUrl = vsprintf('%s/%s/buckets/%s/submissions',
+            [SubmissionsApi::ENDPOINT_URL, $this->projectId, $bucketName]);
+
+
+        $requestStructure = [
+            'headers' => [
+                'Accept' => 'application/json',
+                'Authorization' => vsprintf('%s %s',
+                    [$this->authProvider->getTokenType(), $this->authProvider->getAccessToken()]),
+            ],
+            'exceptions' => false,
+            'json' => $testRequestBody
+        ];
+
+        $this->client
+            ->expects(self::once())
+            ->method('request')
+            ->with('post', $endpointUrl, $requestStructure)
+            ->willReturn($this->responseMock);
+
+        $response = $this->object->createSubmission($bucketName, $testRequestBody);
+        self::assertEquals($testExpectedResponse, $response);
```

Comment:
Looks strange. You assert what you mocked. I mean `$testExpectedResponse` is the same as `$testRawResponse`. Let's get rid of this redundant assertion here and throughout the tests. Therefore, we don't need `mockClientResponse()` method anymore. In the tests, we only need to assert input parameters and request data we pass to the `request()` method.

Reply:
I tested that request that i send is like it should be AND response is correctly parsed

Reply:
Ok, then you test the same logic (`BaseApiAbstract::sendRequest()`) in all your tests. This should be tested in a separate test class. This is not related to submission service tests.

Reply:
Still isn't fixed.